File: braggloopplace_ui/src/services/PurchaseService.js
import http from "../http-common";

class PurchaseService {
  getAllPurchases(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/purchase/purchases`, searchDTO);
  }

  get(purchaseId) {
    return this.getRequest(`/purchase/${purchaseId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/purchase?field=${matchData}`, null);
  }

  addPurchase(data) {
    return http.post("/purchase/addPurchase", data);
  }

  update(data) {
  	return http.post("/purchase/updatePurchase", data);
  }
  
  uploadImage(data,purchaseId) {
  	return http.postForm("/purchase/uploadImage/"+purchaseId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PurchaseService();

File: src/main/java/com/braggloopplace/service/PurchaseService.java
package com.braggloopplace.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggloopplace.domain.Purchase;
import com.braggloopplace.dto.PurchaseDTO;
import com.braggloopplace.dto.PurchaseSearchDTO;
import com.braggloopplace.dto.PurchasePageDTO;
import com.braggloopplace.dto.PurchaseConvertCriteriaDTO;
import com.braggloopplace.service.GenericService;
import com.braggloopplace.dto.common.RequestDTO;
import com.braggloopplace.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PurchaseService extends GenericService<Purchase, Integer> {

	List<Purchase> findAll();

	ResultDTO addPurchase(PurchaseDTO purchaseDTO, RequestDTO requestDTO);

	ResultDTO updatePurchase(PurchaseDTO purchaseDTO, RequestDTO requestDTO);

    Page<Purchase> getAllPurchases(Pageable pageable);

    Page<Purchase> getAllPurchases(Specification<Purchase> spec, Pageable pageable);

	ResponseEntity<PurchasePageDTO> getPurchases(PurchaseSearchDTO purchaseSearchDTO);
	
	List<PurchaseDTO> convertPurchasesToPurchaseDTOs(List<Purchase> purchases, PurchaseConvertCriteriaDTO convertCriteria);

	PurchaseDTO getPurchaseDTOById(Integer purchaseId);







}





